app/controllers/public/customers_controller.rb

```diff
@@ -1,16 +1,36 @@
 class Public::CustomersController < ApplicationController
   def show
+    @customer = current_customer
   end
 
   def edit
+    @customer = current_customer
   end
 
   def update
+    @customer = current_customer
+    if @customer.update(customer_params)
+      redirect_to my_page_customers_path
+    else
+      render :edit
+    end
   end
 
   def out
+    @customer = current_customer
+    @customer.update(is_deleted: true)
+    reset_session
+    redirect_to root_path
   end
 
   def out_confirm
   end
+  
+  
+  
+  private
+  def customer_params
+    params.require(:customer).permit(
+      :first_name, :last_name, :first_name_kana, :last_name_kana, :postal_code, :address, :tel, :email, :is_deleted)
```

Comment:
編集フォームからの送信情報に`:is_deleted`は含まれないので不要です。
```suggestion
      :first_name, :last_name, :first_name_kana, :last_name_kana, :postal_code, :address, :tel, :email)
```